pkg/deps/config/config.go

```diff
@@ -37,36 +42,129 @@ func SupplyConfig(
 }
 
 // NewSupplyEventsQueryClientFn returns a new function which constructs an
-// EventsQueryClient instance and returns a new depinject.Config which is supplied
-// with the given deps and the new EventsQueryClient.
+// EventsQueryClient instance and returns a new depinject.Config which is
+// supplied with the given deps and the new EventsQueryClient.
 func NewSupplyEventsQueryClientFn(
-	pocketNodeWebsocketUrl string,
+	pocketNodeWebsocketURL string,
 ) SupplierFn {
 	return func(
 		_ context.Context,
 		deps depinject.Config,
 		_ *cobra.Command,
 	) (depinject.Config, error) {
-		eventsQueryClient := eventsquery.NewEventsQueryClient(pocketNodeWebsocketUrl)
+		eventsQueryClient := eventsquery.NewEventsQueryClient(pocketNodeWebsocketURL)
 
 		return depinject.Configs(deps, depinject.Supply(eventsQueryClient)), nil
 	}
 }
 
-// NewSupplyBlockClientFn returns a function which constructs a BlockClient instance
-// with the given nodeURL and returns a new depinject.Config which is supplied with
-// the given deps and the new BlockClient.
-func NewSupplyBlockClientFn(pocketNodeWebsocketUrl string) SupplierFn {
+// NewSupplyBlockClientFn returns a function which constructs a BlockClient
+// instance with the given nodeURL and returns a new depinject.Config which
+// is supplied with the given deps and the new BlockClient.
+func NewSupplyBlockClientFn(pocketNodeWebsocketURL string) SupplierFn {
 	return func(
 		ctx context.Context,
 		deps depinject.Config,
 		_ *cobra.Command,
 	) (depinject.Config, error) {
-		blockClient, err := block.NewBlockClient(ctx, deps, pocketNodeWebsocketUrl)
+		blockClient, err := block.NewBlockClient(ctx, deps, pocketNodeWebsocketURL)
 		if err != nil {
 			return nil, err
 		}
 
 		return depinject.Configs(deps, depinject.Supply(blockClient)), nil
 	}
 }
+
+// NewSupplyQueryClientContextFn returns a function with constructs a ClientContext
+// instance with the given cmd and returns a new depinject.Config which is
+// supplied with the given deps and the new ClientContext.
+func NewSupplyQueryClientContextFn(pocketQueryNodeURL string) SupplierFn {
+	return func(_ context.Context,
+		deps depinject.Config,
+		cmd *cobra.Command,
+	) (depinject.Config, error) {
+		// Set --node flag to the --pocket-node for the client context
+		// This flag is read by cosmosclient.GetClientQueryContext.
+		err := cmd.Flags().Set(cosmosflags.FlagNode, pocketQueryNodeURL)
```

Comment:
Given the apparent design of parts of the cosmos-sdk and that they seem to effectively rely on the shared mutable state of the command flags, perhaps we should consider either resetting this value to empty string or whatever it was before we modified it. Wdyt?